File: backend/fooddelivery_backend/src/main/java/com/fooddelivery/orderservice/entity/Order.java
package com.fooddelivery.orderservice.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "customer_orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_email")
    private String userEmail;

    private Double totalAmount;
    private String status; // PENDING, SUCCESS, CANCELLED
    private String paymentMethod;

    @Column(name = "delivery_address")
    private String deliveryAddress;

    @Column(name = "order_date")
    private LocalDateTime orderDate;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "order_id")
    private List<OrderItem> items;
}

File: backend/AdminRiderService/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdminRiderService.Models
{
    [Table("customer_orders")]
    public class Order
    {
        [Column("id")]
        public long Id { get; set; }

        [Column("user_email")]
        [Required]
        public string? UserEmail { get; set; }

        [Column("status")]
        [Required]
        public string? Status { get; set; } // "Placed", "Preparing", "Out for Delivery", "Delivered"

        [Required]
        [Column("total_amount", TypeName = "decimal(10,2)")]
        public decimal TotalAmount { get; set; }

        [Column("order_date")]
        public DateTime OrderDate { get; set; } = DateTime.Now;

        [Column("payment_method")]
        public string? PaymentMethod { get; set; }

        [Column("delivery_address")]
        public string? DeliveryAddress { get; set; }

        public List<OrderItem> Items { get; set; } = new List<OrderItem>();
    }
}
